pkg/staticnetworkconfig/generator.go

```diff
@@ -197,21 +195,14 @@ func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistence(macInterf
 					s.log.Infof("Interface %s has no mac-interface mapping but matches a physical interface", interfaceName.String())
 				}
 			}
-			matched = true
 		case "vlan":
 			vlanSection := cfg.Section("vlan")
-			parent, err := vlanSection.GetKey("parent")
+			_, err := vlanSection.GetKey("parent")
 			if err != nil {
 				return errors.Wrapf(err, "failed to get parent of vlan %s", interfaceName.Name())
```

Comment:
We're no longer doing any real validation with the VLANs, so we probably don't really need to look at it them at all.

Reply:
Removed vlan check.

Reply:
@zaneb  @bfournie  In that case, users can provide a YAML with a VLAN where the base interface is a logical name, and with no mapping of the name associated with a MAC address. Why don’t we want a validation for this scenario?

Reply:
They could also have done that before, because this validation wasn't checking for that. All it did was make sure that an entry in the Name/MAC mapping counted even if the interface wasn't defined in the NMState but a VLAN based on it was.

You _could_ validate something interesting about VLANs - that the parent interface of the VLAN is either defined in the NMState (this would require two passes through the list), is defined in the Name/MAC mapping list, or has a plausible physical name. That may well be worth doing as a separate ticket. But the code being removed did not validate anything comparable to that.

Reply:
But previously, if the base interface wasn't included in the mapping, `matched = false`, and the error "no mac address mapping can be associated with the available network interfaces" would be raised, isn't it?